src/modules/configurator/components/ConfigPropertyDropdown.tsx

```diff
@@ -25,8 +33,31 @@ export function ConfigPropertyDropdown({ propertyName, isActive }: Props) {
     }
   };
 
+  const handleSetActivePropIndex = () => {
+    if (!configurations || !userConfiguration) return;
+
+    const index = configurations[propertyName].findIndex(
+      (item) => item.label === userConfiguration[propertyName]?.label
+    );
+
+    setActivePropIndex((prevActivePropIndex) => ({
+      ...prevActivePropIndex,
+      [propertyName]: index,
+    }));
+  };
+
+  useEffect(() => {
+    if (configurations && userConfiguration) handleSetActivePropIndex();
```

Comment:
You have the check inside the fucntion, also maybe wrap again with useCallback.

Reply:
Wrapped in a useCallback.